React and API integration basic

sachsom95 opened the pull request:
Main Review

Commit:
Create Django api app

Commit:
- Register App
- Create Base URL

Commit:
Add request in requirments
- Add api url for aliases

Commit:
- Add get_domain
- Add get_aliases

Commit:
Add all GET Request and GET Endpoints

Commit:
Add Environment variable template
- Add environ library
- Modified manage.py to load env files on start
- Add env template
- improved ignore file

sachsom95 committed:
Merge pull request #4 from MLH-Sprint-2/sachin_basic_api

Sachin basic api

Commit:
Make Base Alias Table

Commit:
Add Aliases Model
- created __str__ for Aliases model
- Add to admin
- add Allowed Host for Admin to Settings

Commit:
- Dummy Sqlite table

LucasLobo commented on app/.env-template:
```diff
@@ -0,0 +1,2 @@
+Rename the file to .env remove this line and add keyvalue pairs like below
+PASSWORD=enter_password_here
```
This should be USERNAME=

LucasLobo commented on app/api/views.py:
```diff
@@ -0,0 +1,55 @@
+import os
+# from rest_framework.response import Response
+
+PASSWORD = os.environ.get('PASSWORD')
```
USERNAME=

LucasLobo commented on app/api/views.py:
```diff
@@ -0,0 +1,55 @@
+
+def request_get_util():
+    return requests.get(FORWARD_EMAIL_ENDPOINT, auth=(PASSWORD, ''))
+
```
auth=(USERNAME, '')

LucasLobo commented on app/api/views.py:
```diff
@@ -0,0 +1,55 @@
+PASSWORD = os.environ.get('PASSWORD')
+DOMAINS_PROVIDED = {"domains": ["hash.fyi", "hideaddress.net",
+                                "mailsire.com", "secret.fyi"]
+                    }
```
This is missing `swiftmegaminds.tech`

LucasLobo commented on app/api/urls.py:
```diff
@@ -0,0 +1,10 @@
+from . views import get_aliases, get_domains, get_alias_filtered
+urlpatterns = [
+    path('aliases', get_aliases, name="get_aliases"),
+    path('aliases/domains', get_domains, name="get_domains"),
```
`path('aliases/domains', get_domains, name="get_domains"),`
should be
`path('domains', get_domains, name="get_domains"),`

LucasLobo replied:
We don't really require this endpoint either

LucasLobo commented on app/api/urls.py:
```diff
@@ -0,0 +1,10 @@
+from django.urls import path, include
+from . views import get_aliases, get_domains, get_alias_filtered
+urlpatterns = [
+    path('aliases', get_aliases, name="get_aliases"),
```
`alias `instead of `aliases`

sachsom95 replied:
can you confirm the endpoint should have api/alias?

LucasLobo replied:
It should be /api/v1/alias

sachsom95 replied:
changed

LucasLobo commented on app/api/urls.py:
```diff
@@ -0,0 +1,10 @@
+    path('aliases', get_aliases, name="get_aliases"),
+    path('aliases/domains', get_domains, name="get_domains"),
+    path('aliases/domains/<str:DOMAIN>',
+         get_alias_filtered, name="get_alias_filtered"),
```
`'aliases/domains/<str:DOMAIN>' ` should be `'alias/<str:DOMAIN>'`

LucasLobo commented on app/api/views.py:
```diff
@@ -0,0 +1,55 @@
+                                "mailsire.com", "secret.fyi"]
+                    }
+FORWARD_EMAIL_ENDPOINT = "https://api.forwardemail.net/v1/domains/hideaddress.net/aliases"
+
```
This endpoint should not have the address hard-coded

LucasLobo commented on app/manage.py:
```diff
@@ -3,8 +3,16 @@
+    )
+    environ.Env.read_env()
+
+    print(f"The env loaded : {os.environ.get('PASSWORD')}")
```
'USERNAME' instead of 'PASSWORD'

Commit:
- Changes .env to have USERNAME

Commit:
- Change manage.py to have env as USERNAME from PASSWORD

Commit:
- change URLS from review

Commit:
- Resolve review issue

Commit:
- Remove hardcoding of data

Commit:
- Remove typo and wrong assignments

Commit:
Add .env to gitignore

Commit:
Add .env to gitignore

Commit:
Rebase DB to staging

Merge branch 'staging' of https://github.com/MLH-Sprint-2/MailSafe-api into staging

Commit:
Add Serializer class for Aliases Model

Commit:
- Add ViewSet class for Aliases

Commit:
Merge branch 'staging' of https://github.com/MLH-Sprint-2/MailSafe-api into staging

sachsom95 committed:
Cors policy (#11)

* -Add Cors package to requirement.txt

* - Add CORS to Installed apps

* - Add Cors middleware above CommonMiddleware as per Docs

* - Allowed port 3000 and 8000 for react

Commit:
Correct Typo in CORS

Commit:
increased number of allowed host in test env

LucasLobo commented on app/api/views.py:
```diff
@@ -0,0 +1,73 @@
+    data = []
+    for x in res.json():
+        if x["domain"]["name"] == DOMAIN:
+            data.append(x)
```
This for loop is not required since the domain will already be DOMAIN. The returned data should have the following format:
```
{
    "name": "mlh",
    "domain": "swiftmegaminds.tech",
    "id": "2131d12d121d",
    "recipients": [
        "xxx@gmail.com"
    ],
    "is_enabled": true
},
```

sachsom95 replied:
Resolved tested with react

Commit:
Amend Fetch Functionality to work with React
